### PR TITLE
fix: get the realm config from the service when building the CR

### PR DESCRIPTION
**Pull request by ziccardi**
## Description
We have 2 realms configured into the KeycloakService config. Only the service knows which one is the one to be used, so the realm config have to be requested to the service.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has created for changes required on the client side

### CODE REVIEW

**Review by akoserwal — APPROVED** [review]

**Comment by akoserwal** [comment]
@ziccardi Please can you also update this: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/internal/services/kas_fleetshard_operator_addon.go#L160

**Review by akoserwal — APPROVED** [review]

File: internal/kafka/internal/services/kafka.go [hunk 1]
```diff
@@ -940,13 +939,16 @@ func BuildManagedKafkaCR(kafkaRequest *dbapi.KafkaRequest, kafkaConfig *config.K
 		Status: managedkafka.ManagedKafkaStatus{},
 	}
 
+	keycloakConfig := keycloakService.GetConfig()
+	keycloakRealmConfig := keycloakService.GetRealmConfig()
```
Comment (machi1990) [inline internal/kafka/internal/services/kafka.go:943]: The `GetRealmConfig` method might be too generic. Which realm? The service accounts realm or the sre realm?

Comment (ziccardi) [reply internal/kafka/internal/services/kafka.go:943]: Not sure I get it.
Currently in the `KafkaServiceConfiguration` we have 3 realms configurations:

- KafkaRealm
- OSDClusterIDPRealm
- RedhatSSORealm

However a single KeycloackService instance will use only one of them at a time. The `GetRealmConfig` method of the `KeycloakService` object returns the `RealmConfig` that the current `KeyclocakService` instance is using.

Comment (machi1990) [reply internal/kafka/internal/services/kafka.go:943]: > However a single KeycloakService instance will use only one of them. 

You are write. I thought that was changed but I was wrong.

> The GetRealmConfig method of the KeycloakService object returns the RealmConfig that the current KeyclocakService instance is using.

I agree. I see where I my reasoning was wrong. Thanks for the clarification.

**Pull request merged**